test/basictests.jl

```diff
@@ -349,10 +349,10 @@ end
     end
 
     @testset "Solve Function" begin
-        A1 = rand(n) |> Diagonal
+        A1 = rand(n) |> Diagonal |> Array
         b1 = rand(n)
         x1 = zero(b1)
-        A2 = rand(n) |> Diagonal
+        A2 = rand(n) |> Diagonal |> Array
```

Comment:
If it's not working on v1.6, the answer isn't to remove the test 😅

Reply:
Just make the direct ldiv version require a newer Julia version

Reply:
reverted this change. `lu` isn't defined for `Diagonal` in `1.6` so I just did `lu(Array(D))` haha

Reply:
In the highlighted lines, `A1, A2` need to be `Diagonal` becuase I am doing `ldiv!(_,A1, _)` in some of the tests

Reply:
bump the version required for the ldiv! test

Reply:
It's not necessary. I reverted the change in these lines. So `A1, A2` are `Diagonal` (as before) and the test with `ldiv!` in `@testset "Linear Solve Function"` remains the same. I am using the same `A1, A2` in test for `DirectLdiv!` but was doing `F = lu(A)` which was failing in `1.6.7` because `A isa Diagonal`. So the only change I needed to make was `F = lu(Array(A))`.

Reply:
in fact, i took a second look at the test and I realized I dont need to do a factorization. So I'll remove that now